src/terrain/terrain_mesh.cpp

```diff
@@ -7,9 +7,30 @@
 
 namespace terrain {
 
+TerrainMesh::TerrainMesh(entity::Mesh mesh) {
+    init(mesh.indices_,
+         mesh.indexed_vertices_,
+         mesh.indexed_colors_,
+         mesh.indexed_normals_);
+}
+
 TerrainMesh::TerrainMesh(
-    std::vector<unsigned short>& indices, std::vector<glm::vec3>& indexed_vertices,
-    std::vector<glm::vec3>& indexed_colors, std::vector<glm::vec3>& indexed_normals
+    const std::vector<unsigned short>& indices,
+    const std::vector<glm::vec3>& indexed_vertices,
+    const std::vector<glm::vec3>& indexed_colors,
+    const std::vector<glm::vec3>& indexed_normals
+) {
+    init(indices,
+         indexed_vertices,
+         indexed_colors,
+         indexed_normals);
+}
+
+void TerrainMesh::init(
+    const std::vector<unsigned short>& indices,
+    const std::vector<glm::vec3>& indexed_vertices,
+    const std::vector<glm::vec3>& indexed_colors,
+    const std::vector<glm::vec3>& indexed_normals
```

Comment:
Again, why  not fall back to this as a constructor

Reply:
that's not how this works